java/src/jmri/InstanceInitializer.java

```diff
@@ -1,24 +1,53 @@
 package jmri;
 
+import java.util.Set;
+import javax.annotation.Nonnull;
+
 /**
  * Interface providing initialization of specific objects by default. This is
  * used to move references to specific subtypes out of the jmri package and into
- * e.g. jmri.managers.
- * <P>
+ * more specialized packages.
+ * <p>
  * Note that this is only needed when the object can't be created with a
- * no-arguments construtor. In that case, the {@link InstanceManagerAutoDefault}
- * mechanism is a better choice.
- *
- * <P>
+ * no-arguments constructor. In that case, the
+ * {@link InstanceManagerAutoDefault} mechanism is a better choice.
+ * <p>
+ * Instances of this class will normally be used only if they are annotated with
+ * {@code @ServiceProvider(service = InstanceInitializer.class)}
+ * <p>
  * JMRI is distributed in the hope that it will be useful, but WITHOUT ANY
  * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
  * A PARTICULAR PURPOSE. See the GNU General Public License for more details.
- * <P>
+ * <p>
  * @author Bob Jacobsen Copyright (C) 2010
  * @since 2.9.4
  */
 public interface InstanceInitializer {
 
-    public <T> Object getDefault(Class<T> type);
+    /**
+     * Provide a default instance of the given class.
+     * <p>
+     * <strong>Note</strong> calling this method twice for the same class should
+     * not be expected to return the same instance; however, there is no
+     * guarantee that the same instance will not be returned for two calls to
+     * this method.
+     *
+     * @param <T>  the class to get the default for
```

Comment:
(Updated the comment about "Is `<T>` really a @param" so that the `<T>` is visible)